lib/watir/elements/element.rb

```diff
@@ -614,12 +657,14 @@ def assert_is_element(obj)
       end
     end
 
-    def element_call
+    def element_call(exist_check = :wait_for_exists)
+      Wait.timer ||= Wait::Timer.new.tap { |t| t.end_time = ::Time.now + Watir.default_timeout }
+      send exist_check
       yield
     rescue Selenium::WebDriver::Error::StaleElementReferenceError
-      @element = locate
-      assert_element_found
       retry
+    ensure
+      Wait.timer = nil
```

Comment:
Yes, you were right, this didn't do what I thought it was doing. Committed a fix.